Common/src/main/java/com/blamejared/crafttweaker/api/annotation/TaggableElementManagerFactory.java

```diff
@@ -0,0 +1,23 @@
+package com.blamejared.crafttweaker.api.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface TaggableElementManagerFactory {
+    
+    /**
+     * The name of the registry that holds the element that this manager deals with.
+     *
+     * Examples:
+     * - "minecraft:item" for Item
+     * - "minecraft:worldgen/biome" for Biome
+     *
+     * @return The name of the registry that holds the element that this manager deals with.
+     */
+    String value();
```

Comment:
💭 Suggestion:

To make sure people only put in "reasonable" values, we could at least add a vanity check using JetBrain's Pattern annotation
```suggestion
    @Pattern("[a-z\\d_.-]+:[a-z\\d/._-]+")
    String value();
```

Reply:
We're currently not using any intelliJ Annotations right now, and with the current setup the annotations won't be included in the maven pom file, so I don't know if it would really benefit anyone.